run_bot_streaming.sh

```
@@ -0,0 +1,209 @@
+
+    # Get recording and debug settings
+    local recording_mode=${RECORDING:-true}
+    local debug_mode=${DEBUG:-false}
+    local debug_logs=${DEBUG_LOGS:-false}
+
+    print_info "Recording enabled: $recording_mode"
+    print_info "Output directory: $output_dir"
+
+    # Find available port for bot API
+    local main_port=$(find_available_port 3000)
+    if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)